### ISSUE TITLE
How to debug first connection?

### ISSUE THREAD

**Issue by laurentbacall** [barracuda-fsh/pyobd#8]
@barracuda-fsh, many thanks for maintaining and enhancing this cool piece of software. I just received a brand new Chinese ELM327 v1.5. It works just fine on Ubuntu 20.04 by using OBD commands with 'sudo screen /dev/ttyUSB0 38400': I was among others able to retrieve trouble codes.
I can't get it to connect with pyobd with the below error messages. Where should I get started to debug the connection?

Thank you.

[obd.obd] No OBD-II adapters found
[obd.obd] Cannot load commands: No connection to car
[obd.obd] No OBD-II adapters found
[obd.obd] Cannot load commands: No connection to car
[obd.obd] No OBD-II adapters found
[obd.obd] Cannot load commands: No connection to car
[obd.obd] No OBD-II adapters found
[obd.obd] Cannot load commands: No connection to car
[obd.obd] No OBD-II adapters found
[obd.obd] Cannot load commands: No connection to car

**Comment by barracuda-fsh** [comment]
1.) Did you try manually selecting the COM port?
2.) Is the obd adapter plugged into the cars port and the usb of the laptop and is ignition switched to ON? Some adapters don't power on until all these conditions are met.
3.) Other than that, it should work, I have no idea why it wouldn't.

**Comment by laurentbacall** [comment]
Thank you.
1) I did not find an option for this. Only option in the user interface is 'Choose serial port: AUTO'
2) Yes, all three conditions where met when I tested.
4) How would I detect if this is the issue?
5) I patiently tried all possible combination of baud rates and FAST/NORMAL (14 combinations) to no avail.

Please let me know how I can test 1) and 4) further.

**Comment by barracuda-fsh** [comment]
If you can't manually select the port, then the program does not detect the COM port. Hmmmm...... do you have any more debug information in command line or only this that you specified? I will check more on my side tomorrow.

**Comment by laurentbacall** [comment]
Sorry, no more information in terminal output. Happy to try and add any debug code that would help. Just let me know. Thank you.

**Comment by laurentbacall** [comment]
@barracuda-fsh, attempting some manual python3 obd commands, it seems to boil down to a permission issue. Any clue how to solve this one?

********************************
>>> import obd
>>> obd.scan_serial()
[]
********************************

*******************************
>>> connection = obd.OBD("/dev/ttyUSB0")
[obd.obd] ======================= python-OBD (v0.7.1) =======================
[obd.obd] Explicit port defined
[obd.elm327] Initializing ELM327: PORT=/dev/ttyUSB0 BAUD=auto PROTOCOL=auto
[obd.elm327] [Errno 13] could not open port /dev/ttyUSB0: [Errno 13] Permission denied: '/dev/ttyUSB0'
[obd.obd] Closing connection
[obd.obd] Cannot load commands: No connection to car
[obd.obd] ===================================================================
********************************

**Comment by laurentbacall** [comment]
(I tried adding myself to the linux group owning /dev/ttyUSB0 to no avail).

**Comment by laurentbacall** [comment]
It now works -- I needed to log out and back in again after adding myself to the group. Now need to re-try with a car connection. Will do this later today.

>>> import obd
>>> obd.scan_serial()
['/dev/ttyUSB0']
>>> connection = obd.OBD("/dev/ttyUSB0")
Trying baud rate:  38400
Response from baud 38400: b'\x7f\x7f\r?\r\r>'
Choosing baud rate 38400
[obd.elm327] Failed to read port
[obd.elm327] ATE0 did not return 'OK'
[obd.obd] Cannot load commands: No connection to car
>>>

**Comment by laurentbacall** [comment]
So to confirm, it now works fine when connected to the car.
If you wish to update the README.md for linux users, here are the two things I needed to do differently to get it to work:

In the sudo apt install clause, I needed to replace libgstreamer-plugins-base1.0 with libgstreamer-plugins-base1.0-0.

And I needed to issue
sudo usermod -a -G _the-/dev/ttyUSB0-group_ _myUserName_
so that I and python can access the serial connection

Thank you.
